scanpipe/pipes/d2d.py

```diff
@@ -953,3 +953,32 @@ def _map_javascript_colocation_resource(
         from_neighboring_resources,
         map_type="js_colocation",
     )
+
+
+def flag_processed_archives(project):
+    """
+    Resources without an assigned status which are package archives, and all
+    resources inside the archive has a status, should also be considered as
+    processed.
+    """
+    to_resources = project.codebaseresources.files().to_codebase()
+    to_resources_archives = to_resources.no_status().filter(is_archive=True)
+
+    for to_archive in to_resources_archives:
+        archive_extract_path = to_archive.path + "-extract"
+        archive_extract_resource = to_resources.filter(path=archive_extract_path)
+
+        # TODO:There are archives which are not extracted by default, these need to
+        # be accounted for. See https://github.com/nexB/scancode.io/issues/827
+        if not archive_extract_resource:
+            continue
+
+        archive_resources_unmapped = to_resources.no_status().filter(
+            path__startswith=archive_extract_path
+        )
+        # If there are resources in the archives which are unmapped,
+        # they are not considered as processed
+        if archive_resources_unmapped:
+            continue
```

Comment:
I think this is actually not needed and triggers an extra query, ie it evaluates the `archive_resources_unmapped` queryset with a SELECT query then does the UPDATE query.
Where doing directly the UPDATE (that may update nothing) would always ensure only 1 query is executed.
To be confirmed though.

Reply:
I'm not sure I understand what you mean correctly.
The SELECT query and the UPDATE query are on two different set of files, one on `archive_resources_unmapped` and the other on a singular `to_archive`.
The SELECT query is done on all the files in the extracted package, to check if any of them still have no status.
Then if and only there are no files which have status empty in those extracted files, we do an UPDATE on only one resource, which is the resource for the archive in question.
But I can use the `.exists()` function here too as this is same as the case mentioned in https://github.com/nexB/scancode.io/pull/861#discussion_r1295798243 to make this more efficient?

Reply:
Nevermind, I've read too fast thinking the update was applying to the previous queryset.
All good ;)